### PR TITLE
wrote tests for PREP genquery iterators

### PR DESCRIPTION
**Pull request by d-w-moore**
Tested green under irods build and test workflow and Python Rule Engine b/t workflow.
Squash of around 34 subcommits mostly to do with debugging interaction with irods core code

### CODE REVIEW

File: scripts/irods/test/rule_texts_for_tests.py [hunk 1]
```diff
@@ -395,6 +395,25 @@ def acAclPolicy(rule_args, callback, rei):
 '''
 #    ON($userNameClient == "quickshare") { }
 
+##===== Test_Genquery_Iterator =====
```
Comment (trel) [inline scripts/irods/test/rule_texts_for_tests.py]: commented out?  for later?

**Comment by trel** [comment]
code looks good.

please write a nice story in the commit message, you've got all the intermediate messages still in there.

**Comment by trel** [comment]
Can we update the filename from `test_prep_genquery_iter.py` to `test_prep_genquery_iterator.py`?

**Comment by d-w-moore** [comment]
Ok. will make that change and push imminently for approval and test.

**Comment by trel** [comment]
awaiting ub14 passing in CI

**Pull request merged**